### DISCUSSION TITLE
flash size?

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#21] [Q&A]
Which flash size do I need to select for esp32 audio kit?    4mb of 8mb?     by default it is set to 4mb.

**Comment by marcel-licence** [comment]
You can leave this at default (4MB). But you should ensure that PSRAM is enabled.

**Reply by tiretrak** [reply]
> You can leave this at default (4MB). But you should ensure that PSRAM is enabled.

how enable the PSRAM ?

**Reply by marcel-licence** [reply]
![grafik](https://user-images.githubusercontent.com/65245426/172048805-87f34993-49e5-40b2-ac3f-468929bd1fe7.png)
![grafik](https://user-images.githubusercontent.com/65245426/172048828-5e723ab0-67a7-484a-b293-22b908f7c26b.png)

You can also select another ESP32 type which has PSRAM enabled by default like the ESP32 WROVER. Not all types of the ESP32 do support PSRAM

**Reply by tiretrak** [reply]
> 

thank you)

**Reply by tiretrak** [reply]
> 
my esp-wroom-32 don`t have psram (((
![psram](https://user-images.githubusercontent.com/7010744/172053872-e1f9ce80-43e4-4397-8c24-bc6d9c2ceb54.png)

this project working without psram ?

**Reply by marcel-licence** [reply]
It might be possible to use the heap instead. The memory is very small and you might only get short samples or only one sample into the memory.

I could imagine to find a solution

**Reply by marcel-licence** [reply]
I've updated the code, now you can use it also without PSRAM. The memory is very limited but it is enough for some short samples. 
Please look for `SAMPLER_USE_HEAP_ONLY` in config.h and uncomment it.
You can make the value bigger and reduce the delay size if necessary.
You might also need to update the ML_SynthTools library because there was a little issue in the delay.cpp file causing a crash.
